pkg/graph/resolvers/system_intake_grb_presentation_links_test.go

```diff
@@ -21,11 +21,80 @@ func (s *ResolverSuite) TestSetSystemIntakeGRBPresentationLinks() {
 	s.NoError(err)
 	s.NotNil(systemIntake)
 
-	createdLinks := createSystemIntakeGRBPresentationLinkUpload(s, systemIntake.ID)
+	createdLinks := createSystemIntakeGRBPresentationLinkUploadSet(s, systemIntake.ID)
 	s.NotNil(createdLinks)
 }
 
-func createSystemIntakeGRBPresentationLinkUpload(suite *ResolverSuite, systemIntakeID uuid.UUID) *models.SystemIntakeGRBPresentationLinks {
+func (s *ResolverSuite) TestUploadSystemIntakeGRBPresentationDeck() {
+	input := s.createUploadSystemIntakeGRBPresentationDeckInput(s.createUploadSystemIntakeGRBPresentationDeckFileData())
+
+	// Assert insert portion of upsert
+	createdLinks, err := UploadSystemIntakeGRBPresentationDeck(
+		s.testConfigs.Context,
+		s.testConfigs.Store,
+		s.testConfigs.S3Client,
+		input,
+	)
+	s.NoError(err)
+	s.NotNil(createdLinks)
+	s.Nil(createdLinks.ModifiedBy)
+
+	// Assert update portion of upsert
+	updatedLinks, err := UploadSystemIntakeGRBPresentationDeck(
+		s.testConfigs.Context,
+		s.testConfigs.Store,
+		s.testConfigs.S3Client,
+		input,
+	)
+	s.NoError(err)
+	s.NotNil(updatedLinks)
+	s.NotNil(updatedLinks.ModifiedBy)
+}
+
+func (s *ResolverSuite) TestUploadSystemIntakeGRBPresentationDeck_NilFile() {
+	input := s.createUploadSystemIntakeGRBPresentationDeckInput(nil)
+
+	_, err := UploadSystemIntakeGRBPresentationDeck(
+		s.testConfigs.Context,
+		s.testConfigs.Store,
+		s.testConfigs.S3Client,
+		input,
+	)
+	s.NoError(err)
```

Comment:
I might recommend for this test just making sure it returns `nil`, since that's what's in the `links.IsEmpty()` return (that way we're validating that it doesn't even try and create anything)

```suggestion
	createdLinks, err := UploadSystemIntakeGRBPresentationDeck(
		s.testConfigs.Context,
		s.testConfigs.Store,
		s.testConfigs.S3Client,
		input,
	)
	s.NoError(err)
	s.Nil(createdLinks) // since we tried to upload nothing, the resolver will attempt a delete and return nil
```